### DISCUSSION TITLE
Question on RandCropByPosNegLabeld

### DISCUSSION THREAD

**Discussion by RitaHT** [Project-MONAI/MONAI#8128] [Q&A]
Hello! I am using a code passed down from 0.8 era. I am currently on 1.3.2. 
The question happens in the transform, it works fine on my mentor's 0.8 version but not on my version.
It should give (1,128,128,128) but instead it gives (1, 128,128,128) for some images and (1,128,128,127) for others, causing the dataloader to fail. 
I wonder where did it go wrong?

the transforms code is below:
`         

         train_transforms = Compose([
            LoadImaged(keys=["image", "label"]),
            EnsureChannelFirstd(keys=["image", "label"]),
            Orientationd(keys=["image", "label"], axcodes="RAS"),
            ScaleIntensityRanged(keys=["image"], 
                                 a_min=args.a_min, a_max=args.a_max, 
                                 b_min=args.b_min, b_max=args.b_max, clip=True),
            Spacingd(keys=["image", "label"], 
                     pixdim=(args.space_x, args.space_y, args.space_z), 
                     mode=("bilinear", "nearest"), ),
            CropForegroundd(keys=["image", "label"], 
                            source_key="label", 
                            allow_smaller=True), #hardcoded  #will pad automatically?
            SpatialPadd(keys=["image", "label"], 
                        spatial_size=(args.roi_x, args.roi_y, args.roi_z)),
            EnsureTyped(keys=["image", "label"], device=device),
            RandCropByPosNegLabeld(
                    keys=["image", "label"],
                    label_key="label",
                    spatial_size=(args.roi_x, args.roi_y, args.roi_z), #128,128,128
                    pos=1,
                    neg=1,
                    num_samples=1,
                    image_key="image",
                    image_threshold=0), #return a list, need to change the plotting functions

            RandShiftIntensityd(keys=["image"], prob=0.5, offsets=(0.1, 0.2)),
            ToTensord(keys=["image", "label"]),
        ])
`

**Comment by KumoLiu** [comment]
Hi @RitaHT, thanks for your interest here.
Could you please share the data shape before sending to `RandCropByPosNegLabeld`, or if possible share the data then I can take a look at the issue?

BTW, you can add `ResizeWithPadOrCrop` after `RandCropByPosNegLabeld` to ensure the size as a workaround for now.

Thanks.
